### PR TITLE
differentiating required and optional arguments in the help text for …

### PR DESCRIPTION
**Pull request by anusha94**
By default, all the keyword arguments are listed under the **optional arguments** section. This will lead to confusion to the user as to which arguments are _required_ and actually _optional_.

To solve this problem, I have created an argument group called **required arguments** and listed all the required args under this header.

This PR contains modifications to **VMC samples only**.

### CODE REVIEW

File: samples/vmc/networks_nsxt/cgw_firewall_crud.py [hunk 1]
```diff
@@ -39,17 +39,19 @@ def __init__(self):
         parser = argparse.ArgumentParser(
             formatter_class=argparse.ArgumentDefaultsHelpFormatter)
 
-        parser.add_argument('--refresh_token',
+        required_args = parser.add_argument_group('required arguments')
+
+        required_args.add_argument('--refresh_token',
                             required=True,
```
Comment (tianhao64) [inline samples/vmc/networks_nsxt/cgw_firewall_crud.py]: Try to maintain the vertical alignment..

Comment (anusha94) [reply samples/vmc/networks_nsxt/cgw_firewall_crud.py]: Modified in the next commit

File: samples/vmc/networks_nsxt/cgw_firewall_crud.py [hunk 2]
```diff
@@ -39,7 +39,8 @@ def __init__(self):
         parser = argparse.ArgumentParser(
             formatter_class=argparse.ArgumentDefaultsHelpFormatter)
 
-        required_args = parser.add_argument_group('required arguments')
+        required_args = parser.add_argument_group(
```
Comment (pgbidkar) [inline samples/vmc/networks_nsxt/cgw_firewall_crud.py]: required_args is repeated in all the sample files. Can we declare it in library and use?

Comment (anusha94) [reply samples/vmc/networks_nsxt/cgw_firewall_crud.py]: fixed in the next commit

- new `sample_cli.py` file in the `helpers` directory
- all other samples will import `parser`, `required_args` and `optional_args` as needed

File: samples/vsphere/common/sample_cli.py [hunk 3]
```diff
@@ -31,15 +31,17 @@ def build_arg_parser():
     parser = argparse.ArgumentParser(
         description='Standard Arguments for talking to vCenter')
 
-    parser.add_argument('-s', '--server',
+    required_args = parser.add_argument_group(
+            'required arguments')
+    required_args.add_argument('-s', '--server',
                         action='store',
                         help='vSphere service IP to connect to')
```
Comment (pgbidkar) [inline samples/vsphere/common/sample_cli.py]: required=True is missing

File: samples/vmc/networks_nsxv/nat_rule_crud.py [hunk 4]
```diff
@@ -29,35 +29,26 @@ class NatRuleCrud(object):
...
     def __init__(self):
-        parser = argparse.ArgumentParser(
-            formatter_class=argparse.ArgumentDefaultsHelpFormatter)
-        parser.add_argument(
-            '-r',
-            '--refresh-token',
-            required=True,
-            help='VMware Cloud API refresh token')
+        required_args.add_argument(
+            '--org-id', required=True, help='Organization identifier.')
```
Comment (pgbidkar) [inline samples/vmc/networks_nsxv/nat_rule_crud.py]: Most of the samples require org_id and sddc_id. Please check if we can move both to sample_cli.py and build seperate arg parser where it is not required

File: samples/vmc/networks_nsxt/cgw_firewall_crud.py [hunk 5]
```diff
@@ -16,8 +16,8 @@
 __author__ = 'VMware, Inc.'
 __vcenter_version__ = '6.8.0+'
 
-import argparse
 import requests
+from samples.vmc.helpers.sample_cli import parser, required_args
```
Comment (pgbidkar) [inline samples/vmc/networks_nsxt/cgw_firewall_crud.py]: required_args is not required here?

File: samples/vmc/networks_nsxt/dfw_firewall_crud.py [hunk 6]
```diff
@@ -16,8 +16,8 @@
 __author__ = 'VMware, Inc.'
 __vcenter_version__ = '6.8.0+'
 
-import argparse
 import requests
+from samples.vmc.helpers.sample_cli import parser, required_args
```
Comment (pgbidkar) [inline samples/vmc/networks_nsxt/dfw_firewall_crud.py]: required_args is not required here?

File: samples/vmc/networks_nsxt/hello_world.py [hunk 7]
```diff
@@ -15,9 +15,8 @@
 
 __author__ = 'VMware, Inc.'
 
-import argparse
 import pprint
-
+from samples.vmc.helpers.sample_cli import parser, required_args
```
Comment (pgbidkar) [inline samples/vmc/networks_nsxt/hello_world.py]: required_args is not required here?

File: samples/vmc/sddc/sddc_crud.py [hunk 8]
```diff
@@ -34,51 +33,52 @@ class CreateDeleteSDDC(object):
     """
 
     def __init__(self):
-        parser = argparse.ArgumentParser()
+        parser = argparse.ArgumentParser(
+        description='Standard Arguments for talking to vCenter')
```
Comment (pgbidkar) [inline samples/vmc/sddc/sddc_crud.py]: Correct the description

File: samples/vmc/tasks/cancel_task.py [hunk 9]
```diff
@@ -27,19 +26,25 @@
 """
 
 parser = argparse.ArgumentParser(
-            formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+        description='Standard Arguments for talking to vCenter')
```
Comment (pgbidkar) [inline samples/vmc/tasks/cancel_task.py]: Correct the description

File: samples/vmc/tasks/list_tasks.py [hunk 10]
```diff
@@ -28,19 +27,25 @@
...
-                    required=True,
-                    help='Organization identifier.')
-
-parser.add_argument('--task-status',
-                    help='Task status to filter. Possible values are: {} \
-                    Show all tasks if no value is passed'.format(accepted))
+parser = argparse.ArgumentParser(
+        description='Standard Arguments for talking to vCenter')
+
+required_args = parser.add_argument_group(
```
Comment (pgbidkar) [inline samples/vmc/tasks/list_tasks.py]: Importing required_args and populating same name parameter

File: samples/vmc/tasks/list_tasks.py [hunk 11]
```diff
@@ -28,19 +27,25 @@
...
-
-parser.add_argument('--org-id',
-                    required=True,
-                    help='Organization identifier.')
-
-parser.add_argument('--task-status',
-                    help='Task status to filter. Possible values are: {} \
-                    Show all tasks if no value is passed'.format(accepted))
+parser = argparse.ArgumentParser(
+        description='Standard Arguments for talking to vCenter')
```
Comment (pgbidkar) [inline samples/vmc/tasks/list_tasks.py]: correct the description

File: samples/vsphere/common/sample_cli.py [hunk 12]
```diff
@@ -31,15 +31,17 @@ def build_arg_parser():
     parser = argparse.ArgumentParser(
         description='Standard Arguments for talking to vCenter')
 
-    parser.add_argument('-s', '--server',
+    required_args = parser.add_argument_group(
+            'required arguments')
+    required_args.add_argument('-s', '--server',
```
Comment (pgbidkar) [inline samples/vsphere/common/sample_cli.py]: required=True is missing

File: samples/vsphere/common/sample_cli.py [hunk 13]
```diff
@@ -31,15 +31,17 @@ def build_arg_parser():
...
                         action='store',
                         help='vSphere service IP to connect to')
 
-    parser.add_argument('-u', '--username',
+    required_args.add_argument('-u', '--username',
                         action='store',
```
Comment (pgbidkar) [inline samples/vsphere/common/sample_cli.py]: required=True is required=True here :)

**Pull request merged**